### PR TITLE
Add a visualizer based on Pangolin

### PR DESCRIPTION
**Pull request by DEAR18**
Add a visualizer if Pangolin can been found.  Otherwise, do not compile the related code. It works well on my Mac air with the latest version of Pangolin installed. The visualizer shows point cloud, rendered image, gt image, training loss and some other info. It also has a buttom to pause the training process.
visualization

### CODE REVIEW

**Comment by pierotofy** [comment]
Thanks @DEAR18 ! This is a great addition. 🚀 

I noticed a small issue with the display of the ground truth / render images in the viewer, while testing the [banana](https://drive.google.com/file/d/1mUUZFDo2swd6CE5vwPPkjN63Hyf4XyEv/view) dataset, perhaps it has to do with the width/height aspect ratio, but I'm sure it could be fixed with a separate PR.

Screenshot 2024-12-17 at 12 19 31 AM

**Comment by pierotofy** [comment]
p.s. I've added a `-DOPENSPLAT_BUILD_VISUALIZER=ON|OFF` option to CMake in the latest main branch.
